examples/layout_columns/app.py

```diff
@@ -0,0 +1,38 @@
+from kids_say import random_cards
+
+from shiny import App, Inputs, Outputs, Session, reactive, render, ui
+
+app_ui = ui.page_fluid(
+    ui.panel_title(ui.h2("Layout Columns")),
+    ui.input_slider("n_cards", label="Number of Cards", min=1, max=24, value=12),
+    ui.input_action_button("new_cards", "New Cards"),
+    ui.output_ui("layout_columns_example").add_class("mt-3"),
+)
+
+
+def server(input: Inputs, output: Outputs, session: Session):
+    @reactive.calc
+    @reactive.event(input.new_cards, ignore_none=False)
+    def cards():
+        return random_cards(24)
+
+    @render.ui
+    def layout_columns_example():
+        return ui.layout_columns(
+            *cards()[: input.n_cards()],
+            # col_widths=[4, 2, 3, 3],
+            # col_widths={"sm": 3},
+            # col_widths=(8, 4),
+            row_heights=(2, 3, 4),
+            # row_heights="500px",
+            col_widths={
+                "sm": 3,
+                "md": (4, 2, 3, 3),
+                "lg": [3, 4, 2, 3],
+                "xl": (3, 3, 4, 2),
+                "xxl": (2, 3, 3, -1, 3),
+            },
```

Comment:
I think it's more important to show off the "intelligent default behavior" than it is to demonstrate what's possible (I can imagine a "drive by" reader could be off-put by the weird result without reading the source). 

This way, we can still give a hint as to an interesting customization, but have the result show-off the default behavior

```suggestion
            # col_widths=(-2, 4, 4, -2),
            # row_heights=["auto", 1]
```

Reply:
Good point; the current state of this example is not intended as a user demonstration. Where do these examples end up and who is their intended audience? Currently it's aimed at testing and reviewing this PR, and we can certainly clean it up before merging.

Reply:
Ah, I was thinking that this would likely go under `shiny/api-examples/` which is where we put examples designed for the examples section of the API reference. I though that, in order to actually get the example in there, you need to decorate the function with `@add_example`, but it looks like that [isn't necessary for `layout_column_wrap`](https://shiny.posit.co/py/api/ui.layout_column_wrap.html) (btw, can you fix that `width="250%"` -> `width="250px"` in that example)? @schloerke probably knows more about how that works nowadays.

Also, the `./examples/` dir I think is more so intended for 'gallery'-like examples that combine many pieces at once (rather than one function in isolation).

Reply:
Okay, in that case, to not hold up this PR any more than necessary, how about I take this example out before merging and add it in a follow up PR?

Reply:
👍

Reply:
👉 https://github.com/posit-dev/py-shiny/issues/898